File: react-test/src/types/Pokemon.ts
export interface Pokemon {
  id: number
  name: string
  types: PokemonType[] | string[]
  height: number
  width: number
  URL: string
};

export interface PokemonApi {
  name: string
  url: string
  id: number
}

export interface PokemonDetailsApi extends PokemonApi {
  baseExperience: number
  height: number
  weight: number
  abilities: string[]
}

export interface Ability {
  name: string
  url: string
}

export interface PokemonAbilityDetailsApi {
  ability: Ability
  is_hidden: boolean
  slot: number
}

export enum PokemonType {
  Water,
  Fire,
  Rock,
  Other,
}

File: java/Pokemon.java
public class Pokemon {
    private final String name;
    private final int id;
    private final int height;
    private final int weight;

    Pokemon(String name, int id, int height, int weight) {
        this.name = name;
        this.height = height;
        this.id = id;
        this.weight = weight;
    }

    public String getName() {
        return this.name;
    }

    public int getHeight() {
        return this.height;
    }

    public int getId() {
        return this.id;
    }

    public int getWeight() {
        return this.weight;
    }

}
